staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go

```diff
@@ -210,13 +210,24 @@ func (r *proxyHandler) updateAPIService(apiService *apiregistrationapi.APIServic
 	}
 	newInfo.proxyRoundTripper, newInfo.transportBuildingError = restclient.TransportFor(newInfo.restConfig)
 	if newInfo.transportBuildingError == nil && r.proxyTransport != nil && r.proxyTransport.DialContext != nil {
-		switch transport := newInfo.proxyRoundTripper.(type) {
```

Comment:
DialerFor returns transport.DialContext directly.
We need `transport` and override the transport.DialContext element.

Reply:
Introduce this `wrappedTransport` to utilnet?

Reply:
I introduced a new util function ExtractRoundTripper.
Please take a look again.

Reply:
Kindly ping @liggitt for this bug fix.

Without this, kube-aggregator could not work with -v set to larger than 6.

Reply:
re-reading this, is there a reason we don't set Dial on newInfo.restConfig prior to building the transport, rather than trying to extract/fix it up after the fact?

```
	if r.proxyTransport != nil && r.proxyTransport.DialContext != nil {
		newInfo.restConfig.Dial = r.proxyTransport.DialContext
	}
	newInfo.proxyRoundTripper, newInfo.transportBuildingError = restclient.TransportFor(newInfo.restConfig)
	r.handlingInfo.Store(newInfo)
```